woocommerce/woocommerce-android
# Issue/11059 order deletion

Closes: #11059 

⚠️ Depends on https://github.com/wordpress-mobile/WordPress-FluxC-Android/pull/2972

### Description
This PR adds support for deleting orders from the Order Details screen, it uses the changes of the FluxC PR above to handle the deletion with cancellation.
In order to support the tablet mode, I mirrored what was done for products deletion by adding a shared ViewModel `OrdersCommunicationViewModel` that's used to pass the action from the order details screen to the list screen.


### Testing instructions
1. Open order details screen.
2. Notice the trash button.
3. Click on it and notice the alert.
4. Confirm the trash.
5. Notice the Undo snackbar on the order list screen.
6. Repeat 2-5 and use the Undo button of the Snackbar.
7. Confirm the item deletion was reverted.

Repeat the above steps on a tablet.

### Images/gif
[Screen_recording_20240313_101635.webm](https://github.com/woocommerce/woocommerce-android/assets/1657201/279cd2e2-8a7a-4606-8d7a-be343e221d5d)
